src/LuzFaltex.Extensions.DependencyInjection/ServiceLookup/CallSiteFactory.cs

```diff
@@ -0,0 +1,695 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+
+namespace LuzFaltex.Extensions.DependencyInjection.ServiceLookup;
+
+[RequiresDynamicCode(MutableServiceProvider.RequiresDynamicCodeMessage)]
+internal sealed class CallSiteFactory : IServiceProviderIsService
```

Comment:
This file was changed to add support for adding / removing services after the provider is built.